src/clj/rems/api/applications.clj

```diff
@@ -119,23 +119,41 @@
       (update-in [:items] longify-keys)
       (update-in [:licenses] longify-keys)))
 
-;; TODO dynamic events hiding
 (defn- hide-sensitive-events [events]
   (filter (fn [event]
             ((complement contains?) #{"third-party-review" "review-request" "review"} (:event event)))
           events))
 
+(defn- latest-event [events]
+  (apply time/max-date (map #(or (:time %) ; Non-dynamic events
+                                 (:event/time %)) events))) ; Dynamic events
+
+(defn- update-application-last-modified [application]
+  (let [events (or (:events application)
+                   (:dynamic-events application))]
+    (assoc application :last-modified (latest-event events))))
+
+(defn- hide-sensitive-dynamic-events [events]
```

Comment:
This will need to be called from `rems.api.applications-v2/apply-user-permissions`, but to avoid conflicts with #930 it's better to not do it in this PR. Maybe add just a TODO comment to `apply-user-permissions` in this PR and I'll implement it in #930.